### PR TITLE
interleave

### PR DESCRIPTION
**Pull request by jtlap**
interleave to new_style

### CODE REVIEW

File: include/eve/detail/function/interleave.hpp [hunk 1]
```diff
@@ -7,12 +7,30 @@
...
+  {
+
+    template<eve::simd_value T, std::same_as<T>... Ts>
+    EVE_FORCEINLINE kumi::tuple<T, Ts...> operator()(T v, Ts... vs) const noexcept
+    { return EVE_DISPATCH_CALL(v, vs...); }
+
+
+    template<simd_value T, std::same_as<T>... Ts>
+    EVE_FORCEINLINE kumi::tuple<logical<T>, logical<Ts>...> operator()(logical<T> v, logical<Ts>... vs) const noexcept
+    requires( T::abi_type::is_wide_logical )
```
Comment (jfalcou) [inline include/eve/detail/function/interleave.hpp:27]: this should never happen

Comment (jfalcou) [reply include/eve/detail/function/interleave.hpp:27]: the requries is an implemantiond etail, the () must work for all kind of logicals

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/detail/function/interleave.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/detail/function/interleave.hpp:27]: line suppressed

File: include/eve/detail/function/simd/common/interleave.hpp [hunk 2]
```diff
@@ -16,16 +16,16 @@ namespace eve::detail
   // N-ary interleave
   // This is the main one as it is easier to specify optimisation on this interface than on tuple
   //================================================================================================
-  template<simd_value T, std::same_as<T>... Ts>
-  EVE_FORCEINLINE auto interleave_(EVE_SUPPORTS(cpu_),T v0, Ts... vs) noexcept
+  template<callable_options OO, simd_value T, std::same_as<T>... Ts>
+  EVE_FORCEINLINE auto interleave_(EVE_REQUIRES(cpu_), OO const&, T v0, Ts... vs) noexcept
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/function/simd/common/interleave.hpp:20]: Are we passing callable options by const& Is there any reason to? 

I'd expect by value but maybe it's important?

Comment (jtlap) [reply include/eve/detail/function/simd/common/interleave.hpp:20]: I think that this is optimized out anyway here, as only the type is used

Comment (jfalcou) [reply include/eve/detail/function/simd/common/interleave.hpp:20]: I think it is irrelevant as we carry mostly type inforamtions. I guess I was sloppy and jtlap mimicked me.
We can later have a cleaning session to be homogeneous

**Review by jfalcou — APPROVED** [review]

**Pull request merged**